tox.ini

```diff
@@ -63,6 +72,23 @@ commands =
         -m pytest --ignore={[vars]tst_path}integration -v --tb native -s {posargs}
     coverage report
 
+[testenv:coverage-report]
+description = Create test coverage report
+deps =
+    coverage[toml]
+    pytest
```

Comment:
These deps are probably fine unpinned, but most of your Juju and other related deps should be pinned.

Reply:
Ah and I see they are above, so you can ignore the above :)